compiler/lib/code.ml

```diff
@@ -909,15 +909,28 @@ let invariant { blocks; start; _ } =
         assert (not (Var.ISet.mem defs x));
         Var.ISet.add defs x)
     in
+    let check_prim_arg = function
+      | Pc (NativeInt _ | Int32 _) ->
+          assert (
+            match Config.target () with
+            | `Wasm -> true
+            | _ -> false)
+      | Pc _ | Pv _ -> ()
+    in
     let check_expr = function
       | Apply _ -> ()
       | Block (_, _, _, _) -> ()
       | Field (_, _, _) -> ()
       | Closure (l, cont) ->
           List.iter l ~f:define;
           check_cont cont
+      | Constant (NativeInt _ | Int32 _) ->
+          assert (
+            match Config.target () with
```

Comment:
This could be shared with a check_constant helper

Reply:
Done